### PR TITLE
wip

### PR DESCRIPTION
**Pull request by saxonewt**

### CODE REVIEW

File: allschool_test.go [hunk 1]
```diff
@@ -65,9 +67,9 @@ func TestRemoveClass(t *testing.T) {
...
 	var v []openapi.ResponseMemberClass
 	decoder := json.NewDecoder(resp.Body)
-	_ = decoder.Decode(&v)
+	err = decoder.Decode(&v)
+
+	require.Nil(t, err)
 
-	assert.Equal(t, teachers[0], v[0].Owner.Id)
-	assert.Equal(t, members-1, len(v))
-	assert.Equal(t, classes[0], v[0].Id)
```
Comment (anglofrog) [inline allschool_test.go:72]: It does not make sense to remove these, we want to make sure that the classes are what we expect them to be. Also createtestaccounts appears to never add 1 student to 2 classes(no repeat students) so a length of 0 may be right but I feel it makes for an unrealistic test. I also need to add students to the school classes.

**Review by anglofrog — COMMENTED** [review]
This is ok but it does not address why my code was not working and in some cases a bad idea. I think it is because we have been short on time.

File: allschool.go [hunk 2]
```diff
@@ -82,28 +82,30 @@ func (a *AllSchoolApiServiceImpl) RemoveClass(ctx context.Context, body openapi.
...
 			return err
 		}
-		resp = classes
 		return nil
 	})
 	if err != nil {
 		lgr.Printf("ERROR cannot edit class with Id: %s %v", body.Id, err)
 		return openapi.Response(500, "{}"), nil
 	}
+	resp, err := classesWithOwnerDetails(a.db, userDetails.SchoolId, userDetails.Email)
```
[the remark this answers is not in the document]
Comment (saxonewt) [reply allschool.go:105]: this is the only way to implement it with the function
classesWithOwnerDetails opens new read transaction and cannot be used inside other transaction

File: allimpl.go [hunk 3]
```diff
@@ -27,6 +27,31 @@ func UserByIdTx(tx *bolt.Tx, userId string) (user *bolt.Bucket, err error) {
 
 }
 
+func getClassAtSchoolTx(tx *bolt.Tx, schoolId, classId string) (classBucket *bolt.Bucket, err error) {
```
[the remark this answers is not in the document]
Comment (saxonewt) [reply allimpl.go:30]: this is the spec for the function
do you suggest add a comment or rename it?

**Pull request merged**